app/src/main/java/org/mozilla/fenix/tabstray/viewholders/BaseBrowserTabViewHolder.kt

```diff
@@ -36,8 +42,14 @@ abstract class BaseBrowserTabViewHolder(
     ) {
         adapter.registerAdapterDataObserver(OneTimeAdapterObserver(adapter) {
             trayList.scrollToPosition(currentTabIndex)
+            if (adapter.itemCount == 0) {
```

Comment:
Looks great! What do you think about moving this logic to a separate method so that it's cleaner to read? Maybe `updateTrayVisibility(size: Int)` ?